packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -1,12 +1,350 @@
 // SPDX-License-Identifier: AGPL-3.0-only
 pragma solidity ^0.8.0;
 
-import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "./BridgePoolFactoryInterface.sol";
+import "../../../contracts/oracle/interfaces/OptimisticOracleInterface.sol";
+import "../../../contracts/oracle/interfaces/StoreInterface.sol";
+import "../../../contracts/oracle/interfaces/FinderInterface.sol";
+import "../../../contracts/oracle/implementation/Constants.sol";
+import "../../../contracts/common/implementation/AncillaryData.sol";
+import "../../../contracts/common/implementation/Testable.sol";
+import "../../../contracts/common/implementation/FixedPoint.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
 /**
- * @notice Contract deployed on L1 that ingests liquidity from passive liquidity providers and returns them claims to
- * withdraw their funds.
+ * @notice Contract deployed on L1 that provides methods for "Relayers" to fulfill deposit orders that originated on L2.
+ * The Relayers can either post capital to fulfill the deposit instantly, or request that the funds are taken out of
+ * a passive liquidity provider pool following a challenge period. Related, this contract ingests liquidity from
+ * passive liquidity providers and returns them claims to withdraw their funds. Liquidity providers are incentivized
+ * to post collateral by earning a fee per fulfilled deposit order.
+ * @dev A "Deposit" is an order to send capital from L2 to L1, and a "Relay" is a fulfillment attempt of that order.
  */
-abstract contract BridgePool is ERC20 {
+contract BridgePool is Testable {
+    using SafeERC20 for IERC20;
+    using FixedPoint for FixedPoint.Unsigned;
+
+    // Administrative contract that deployed this contract and also houses all state variables needed to relay deposits.
+    BridgePoolFactoryInterface public bridgePoolFactory;
+
+    // A Deposit represents a transfer that originated on an L2 DepositBox contract and can be bridged via this contract.
+    enum DepositState { Uninitialized, PendingSlow, PendingInstant, FinalizedSlow, FinalizedInstant }
+    enum DepositType { Slow, Instant }
+
+    // @dev: There is a limit to how many params a struct can contain. Without encapsulating some of the Deposit params
+    // inside the RelayAncillaryDataContents struct, the compiler throws an error related to this issue:
+    // https://github.com/ethereum/solidity/issues/10930.
+    struct RelayAncillaryDataContents {
+        uint256 depositId;
+        // The following params are inferred by the L2 deposit:
+        address l2Sender;
+        address recipient;
+        uint256 depositTimestamp;
+        address l1Token;
+        uint256 amount;
+        uint256 maxFeePct;
+        uint256 proposerRewardPct;
+        // Relayer will compute the realized fee considering the amount of liquidity in this contract and the pending
```

Comment:
I think this might change marginally based on yesterday's discussion but it's fine for now as this is fee independent.

Reply:
Yeah wasn't sure exactly what we had in mind here.

Reply:
fine to ignore for now. this PR does not factor in this kind of fees and it will need to be a separate process anyway